src/modules/utils/GEN_CSV_READER_fbt.cpp

```diff
@@ -0,0 +1,189 @@
+      writeData(1, *mDOs[1], mDOConns[1]);
+      break;
+    }
+    case scmEventCNFID: {
+      writeData(0, *mDOs[0], mDOConns[0]);
+      writeData(1, *mDOs[1], mDOConns[1]);
+      for(TPortId i = 2; i < getFBInterfaceSpec().mNumDOs; i++){
+        writeData(i, *mDOs[i], mDOConns[i]);
+      }
+      break;
+    }
+    default:
+      break;
+  }
+}
+
+bool GEN_CSV_READER::createInterfaceSpec(const char *paConfigString, SFBInterfaceSpec &paInterfaceSpec) {
+  const char *acPos = strrchr(paConfigString, '_');
+  if(nullptr != acPos){
+    acPos++;
+    paInterfaceSpec.mNumDOs = static_cast<TPortId>(forte::core::util::strtoul(acPos, nullptr, 10) + 2); // we have in addition to the SDs a QI and FILE_NAME data inputs
+
+    mDataOutputNames = new CStringDictionary::TStringId[paInterfaceSpec.mNumDOs];
+    mDataOutputTypeIds = new CStringDictionary::TStringId[paInterfaceSpec.mNumDOs];
+
+    mDataOutputNames[0] = g_nStringIdQO;
+    mDataOutputTypeIds[0] = g_nStringIdBOOL;
+    mDataOutputNames[1] = g_nStringIdSTATUS;
+    mDataOutputTypeIds[1] = g_nStringIdSTRING;
+
+    generateGenericDataPointArrays("RD_", &(mDataOutputTypeIds[2]), &(mDataOutputNames[2]), paInterfaceSpec.mNumDOs - 2);
+
+    //create the interface Specification
+    paInterfaceSpec.mNumEIs = 2;
+    paInterfaceSpec.mEINames = scmEventInputNames;
+    paInterfaceSpec.mNumEOs = 2;
+    paInterfaceSpec.mEONames = scmEventOutputNames;
+    paInterfaceSpec.mDINames = scmDataInputNames;
+    paInterfaceSpec.mDIDataTypeNames = scmDataInputTypeIds;
+    paInterfaceSpec.mNumDOs = 2;
+    paInterfaceSpec.mDONames = mDataOutputNames;
+    paInterfaceSpec.mDODataTypeNames = mDataOutputTypeIds;
+    return true;
+  }
+  return false;
+}
+
+void GEN_CSV_READER::openCSVFile() {
+  QO() = false_BOOL;
+  if(!mCSVFile.is_open()) {
+    mCSVFile.open(FILE_NAME().getStorage().c_str());
+    if(mCSVFile.is_open()) {
+      QO() = true_BOOL;
+      STATUS() = scmOK;
+      DEVLOG_INFO("[GEN_CSV_READER]: File %s successfully opened\n", FILE_NAME().getStorage().c_str());
+    } else {
+      const char* errorCode = strerror(errno); 
+      STATUS() = CIEC_STRING(errorCode, strlen(errorCode));
+      DEVLOG_ERROR("[GEN_CSV_READER]: Couldn't open file %s. Error: %s\n", FILE_NAME().getStorage().c_str(), STATUS().getStorage().c_str());
+    }
+  } else {
+    STATUS() = scmFileAlreadyOpened;
+    DEVLOG_ERROR("[GEN_CSV_READER]: Can't open file %s since it is already opened\n", FILE_NAME().getStorage().c_str());
+  }
+}
+
+void GEN_CSV_READER::closeCSVFile() {
+  QO() = CIEC_BOOL(false);
+  mCSVFile.close();
+  STATUS() = scmOK;
+  DEVLOG_INFO("[GEN_CSV_READER]: File %s successfully closed\n", FILE_NAME().getStorage().c_str());
+}
+
+void GEN_CSV_READER::readCSVFileLine() {
+  if(mCSVFile.is_open()) {
+    std::string line;
+    if(getline(mCSVFile, line)){
+      std::size_t oldPos = 0;
+
+      for(std::size_t i = 0;  i < getNumRD(); i++) {
+        std::size_t newPos = line.find(';', oldPos);
```

Comment:
seeing this, you implicitly assume a German-language CSV file, correct?
English-language (and several others) CSVs use the comma ',' as the separator.
In the current version the use of the CSV Reader is limited to German-language (or others using the semicolon as separator) CSVs. I would suggest to extract the separator to an additional input for the FB, enhancing usability of the FB